File: server-app/models/subject.py
# models/subject.py
from extensions import db

class Subject(db.Model):
    __tablename__ = 'subjects'

    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
    course_id = db.Column(db.Integer, db.ForeignKey('courses.id'), nullable=False)
    name = db.Column(db.String(255), nullable=False)

    def to_dict(self):
        return {
            'id': self.id,
            'course_id': self.course_id,
            'name': self.name
        }

File: go-server/models/subject.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Subject struct {
	ID        uint           `json:"id" gorm:"primaryKey;autoIncrement"`
	Name      string         `json:"name" gorm:"not null;size:255" validate:"required"`
	CourseID  uint           `json:"course_id" gorm:"not null;index" validate:"required"`
	CreatedAt time.Time      `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt time.Time      `json:"updated_at" gorm:"autoUpdateTime"`
	DeletedAt gorm.DeletedAt `json:"-" gorm:"index"`

	// Relationships
	Course   Course    `json:"course,omitempty" gorm:"foreignKey:CourseID"`
	Chapters []Chapter `json:"chapters,omitempty" gorm:"foreignKey:SubjectID"`
}

// ToDict converts subject to map for JSON response
func (s *Subject) ToDict() map[string]interface{} {
	return map[string]interface{}{
		"id":         s.ID,
		"name":       s.Name,
		"course_id":  s.CourseID,
		"created_at": s.CreatedAt.Format(time.RFC3339),
	}
}

// TableName specifies the table name for GORM
func (Subject) TableName() string {
	return "subjects"
}